### PR TITLE
Add unread indicator to the timelineCard header icon

### PR DESCRIPTION
**Pull request by toger5**
Fixes: https://github.com/vector-im/element-web/issues/19635
This uses a lot of the code of the unread Indicator of the pinned messages feature.
It made sense to make them look the same. So it is important to also check how the indicator looks now for the pinned messages since I followed the designs for the maximised widgets
image
image
image
image

Designs:
image

---
Here's what your changelog entry will look like:

## ✨ Features
 * Add unread indicator to the timelineCard header icon ([\#7156](https://github.com/matrix-org/matrix-react-sdk/pull/7156)). Fixes vector-im/element-web#19635 and vector-im/element-web#19635. Contributed by @toger5.

Preview: https://61a5174ccdffc94cc207c6c3--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: res/css/structures/_RightPanel.scss [hunk 1]
```diff
@@ -103,44 +103,59 @@ limitations under the License.
...
+    top: 0;
+    margin: 4px;
+    width: $dot-size;
+    height: $dot-size;
+    border-radius: 50%;
+    transform: scale(1.6);
+    transform-origin: center center;
+    background: rgba($background, 1);
+}
+.mx_RightPanel_headerButton_unreadIndicator {
```
Comment (turt2live) [inline res/css/structures/_RightPanel.scss:128]: newline to make blocks easier to read
```suggestion

.mx_RightPanel_headerButton_unreadIndicator {
```

File: res/css/structures/_RightPanel.scss [hunk 2]
```diff
@@ -103,44 +103,59 @@ limitations under the License.
...
 .mx_RightPanel_pinnedMessagesButton {
     &::before {
         mask-image: url('$(res)/img/element-icons/room/pin.svg');
         mask-position: center;
     }
-
-    .mx_RightPanel_pinnedMessagesButton_unreadIndicator {
+}
+.mx_RightPanel_headerButton_unreadIndicator_bg {
+    content: "";
```
Comment (turt2live) [inline res/css/structures/_RightPanel.scss:116]: is `content` required when not using pseudo elements?

File: res/css/structures/_RightPanel.scss [hunk 3]
```diff
@@ -103,44 +103,59 @@ limitations under the License.
...
+    transform: scale(1);
+    background: rgba($pulse-color, 1);
+    box-shadow: 0 0 0 0 rgba($pulse-color, 1);
+    animation: mx_RightPanel_indicator_pulse 2s infinite;
+    animation-iteration-count: 1;
+    &.mx_Indicator_gray {
+        background: rgba($input-darker-fg-color, 1);
+        box-shadow: rgba($input-darker-fg-color, 1);
+    }
+    &::after {
```
Comment (turt2live) [inline res/css/structures/_RightPanel.scss:145]: block legibility:
```suggestion
    animation-iteration-count: 1;

    &.mx_Indicator_gray {
        background: rgba($input-darker-fg-color, 1);
        box-shadow: rgba($input-darker-fg-color, 1);
    }

    &::after {
```

File: src/components/views/right_panel/RoomHeaderButtons.tsx [hunk 4]
```diff
@@ -44,6 +46,12 @@ const ROOM_INFO_PHASES = [
     RightPanelPhases.EncryptionPanel,
     RightPanelPhases.Room3pidMemberInfo,
 ];
+const UnreadIndicator = ({ className }) => {
```
Comment (turt2live) [inline src/components/views/right_panel/RoomHeaderButtons.tsx:49]:
```suggestion

const UnreadIndicator = ({ className }) => {
```

Comment (turt2live) [inline src/components/views/right_panel/RoomHeaderButtons.tsx:49]: need to define types on `className`

File: src/components/views/right_panel/RoomHeaderButtons.tsx [hunk 5]
```diff
@@ -69,19 +77,25 @@ const PinnedMessagesHeaderButton = ({ room, isHighlighted, onClick }) => {
...
-        unreadIndicator = <div className="mx_RightPanel_pinnedMessagesButton_unreadIndicator" />;
+    switch (RoomNotificationStateStore.instance.getRoomState(room).color) {
+        case NotificationColor.Grey:
+            unreadIndicator =
+            <UnreadIndicator className="mx_RightPanel_headerButton_unreadIndicator mx_Indicator_gray" />;
+            break;
+        case NotificationColor.Red:
+            unreadIndicator =
+            <UnreadIndicator className="mx_RightPanel_headerButton_unreadIndicator" />;
+            break;
```
Comment (turt2live) [inline src/components/views/right_panel/RoomHeaderButtons.tsx:89]: the indentation of these case statement bodies is unclear

File: src/components/views/right_panel/RoomHeaderButtons.tsx [hunk 6]
```diff
@@ -69,19 +77,25 @@ const PinnedMessagesHeaderButton = ({ room, isHighlighted, onClick }) => {
...
     }
-
     return <HeaderButton
         name="timelineCardButton"
         title={_t("Chat")}
         isHighlighted={isHighlighted}
         onClick={onClick}
-        analytics={["Right Panel", "Timeline Panel Button", "click"]}
-    >
+        analytics={["Right Panel", "Timeline Panel Button", "click"]}>
```
Comment (turt2live) [inline src/components/views/right_panel/RoomHeaderButtons.tsx:98]: The `>` being on a new line is deliberate for readability

File: src/components/views/rooms/RoomHeader.tsx [hunk 7]
```diff
@@ -98,7 +103,9 @@ export default class RoomHeader extends React.Component<IProps, IState> {
         // redisplay the room name, topic, etc.
         this.rateLimitedUpdate();
     };
-
```
Comment (turt2live) [inline src/components/views/rooms/RoomHeader.tsx:101]: please restore this blank line, and add one after the function declaration

**Review by turt2live — CHANGES_REQUESTED** [review]
The code seems fine - just a bunch of manual linting. Thanks!

**Review by gaelledel — APPROVED** [review]

File: src/components/views/right_panel/RoomHeaderButtons.tsx (hunk 4 above)
Comment (toger5) [reply src/components/views/right_panel/RoomHeaderButtons.tsx]: I am not sure If the changes I did is what you requested here. Let me know if you expected sth. else

File: src/components/views/right_panel/RoomHeaderButtons.tsx [hunk 8]
```diff
@@ -45,15 +47,30 @@ const ROOM_INFO_PHASES = [
     RightPanelPhases.Room3pidMemberInfo,
 ];
 
-const PinnedMessagesHeaderButton = ({ room, isHighlighted, onClick }) => {
+type UnreadIndicatorProps = {className: string};
+
+const UnreadIndicator = ({ className }: UnreadIndicatorProps) => {
```
Comment (t3chguy) [inline src/components/views/right_panel/RoomHeaderButtons.tsx:52]: props should be interfaces

```suggestion
interface IUnreadIndicatorProps {
    className: string;
}

const UnreadIndicator = ({ className }: IUnreadIndicatorProps) => {
```

Comment (toger5) [reply src/components/views/right_panel/RoomHeaderButtons.tsx:52]: Thanks!

**Review by turt2live — APPROVED** [review]
Thanks!

**Comment by Palid** [comment]
@toger5 seems ready to merge, do the honors! 😄

**Pull request merged**